app/client/src/workers/Lint/utils.ts

```diff
@@ -113,3 +153,11 @@ const lintErrorFilters = (lintError: LintError) => {
   }
   return true;
 };
+
+export const addAppsmithFunctionsKeysToDataTree = (dataTree: DataTree) => {
+  const globalData: Record<string, unknown> = Object.assign({}, dataTree);
+  Object.keys(DATA_TREE_FUNCTIONS).forEach((func) => {
+    globalData[func] = true;
```

Comment:
We will need to check for `funcOrFuncCreator` similar to how we do in `enhanceDataTreeWithFunctions` but here we want to avoid all action which has qualifier.
This will avoid to add `run`, `clear` , `getGeoLocation`,  `watchGeoLocation`, `stopWatchGeoLocation` which we need to avoid here. 
```suggestion
if (
      !(typeof funcOrFuncCreator === "object" &&
      "qualifier" in funcOrFuncCreator)
    ) {
    globalData[func] = true;
    }
```

Reply:
@Rishabh-Rathod Nice suggestion, one caveat is that in enhanceDataTreeWithFunctions, we are doing a deep clone.  That is why I am using a new function for this.